pkg/config/static/entrypoint_getlistenconfig_unix.go

```diff
@@ -0,0 +1,36 @@
+//go:build linux || freebsd || openbsd || darwin
+
+package static
+
+import (
+	"net"
+	"syscall"
+
+	"golang.org/x/sys/unix"
+)
+
+// GetListenConfig returns a net.ListenConfig for the address field of the entry
+// point based on its reuse port field.
+func (ep EntryPoint) GetListenConfig() net.ListenConfig {
+	if !ep.ReusePort {
+		return net.ListenConfig{}
+	}
+	return net.ListenConfig{
+		Control: func(_, _ string, c syscall.RawConn) error {
+			var err error
+			if controlErr := c.Control(func(fd uintptr) {
+				err = unix.SetsockoptInt(int(fd), unix.SOL_SOCKET, unix.SO_REUSEADDR, 1)
+				if err != nil {
+					return
+				}
+				err = unix.SetsockoptInt(int(fd), unix.SOL_SOCKET, unixSOREUSEPORT, 1)
+				if err != nil {
+					return
+				}
```

Comment:
I recommend reading https://stackoverflow.com/a/14388707/5801959, it is a very detailed answer.

To put it simply, due to various historical reasons and differences in how different kernels implement them, we must set both of them at the same time to get what we want.

Reply:
@aofei Thanks for this explanation.
However, I'm wondering why we couldn't make only use of `SO_REUSEPORT`.
What was, regarding the `reusePort` feature in Traefik, the rationale for using `SO_REUSEADDR`?
If it was for <3.9 kernels support, the `SetsockoptInt` call with `SO_REUSEPORT` would likely return an error, so I guess the option would not work well with old kernels anyway.

Reply:
Setting `SO_REUSEPORT` without `SO_REUSEADDR` may lead to issues like inability to immediately rebind a server to a port if the previous instance left sockets in the `TIME_WAIT` state.

And from https://man7.org/linux/man-pages/man7/socket.7.html:

> SO_REUSEPORT (since Linux 3.9)
>
> Permits multiple AF_INET or AF_INET6 sockets to be bound
> to an identical socket address.
> …

If only `SO_REUSEPORT` is set, we cannot bind two sockets to `0.0.0.0:80` and `192.168.1.2:80`, respectively, because `SO_REUSEPORT` typically requires all sockets to bind to an identical socket address. However, this behavior can vary with different kernel implementations. For example, Darwin does not impose such a restriction.

---

Despite the above issues, it appears we can safely omit setting `SO_REUSEADDR` in Traefik, as `net.ListenConfig` already [sets it by default](https://github.com/golang/go/blob/923ab13f9bb6fccc362d5259e2f456abbeda0c51/src/net/sockopt_linux.go#L26-L35). Sorry I just found out about this today.

So, I think we can remove the code that sets `SO_REUSEADDR` and leave a comment referencing https://go.dev/src/net/sockopt_linux.go. What do you think @rtribotte?

Reply:
@aofei Good point indeed for the `TIME_WAIT` state.
I'm not sure about the wildcard IPs by looking at the output of the [code](https://rextester.com/BUAFK86204) shared in the Stackoverflow post.
It looks like when `SO_REUSEPORT` is set on both sockets, it does not require all sockets to bind to an identical socket address.
But it is surely depending on the kernel implementation.

Yes, since `net.ListenConfig` does already set it by default, and does it accordingly to the OS, I think it is better to only set `SO_REUSEPORT`.

Thanks!

Reply:
I'm marking the PR as waiting for corrections.